effekt/shared/src/main/scala/effekt/machine/Transformer.scala

```diff
@@ -54,18 +55,19 @@ object Transformer {
       Include(contents)
   }
 
-  def transform(stmt: lifted.Stmt)(using BlocksParamsContext, Context): Statement =
+  def transform(stmt: lifted.Stmt)(using BlocksParamsContext, ModuleContext, Context): Statement =
     stmt match {
       case lifted.Scope(definitions, rest) =>
 
         definitions.foreach {
-          case Definition.Def(id, tpe, block @ lifted.BlockLit(params, body)) =>
+          case Definition.Def(id, tpe, block @ lifted.BlockLit(tparams, params, body)) =>
             // TODO does not work for mutually recursive local definitions
             val freeParams = lifted.freeVariables(block).toList.collect {
-              case id: symbols.ValueSymbol => Variable(transform(id), transform(tpe))
-              case id: symbols.BlockParam => Variable(transform(id), transform(tpe))
-              case id: symbols.ResumeParam => Variable(transform(id), transform(tpe))
-              case id: lifted.EvidenceSymbol => Variable(transform(id), builtins.Evidence)
+              case lifted.ValueParam(id, tpe) => Variable(transform(id), transform(tpe))
+              case lifted.BlockParam(id: (symbols.BlockParam | symbols.ResumeParam), tpe) =>
```

Comment:
Beautiful, much better!

Reply:
Why do you need the union type here? Can't you just drop it?

Reply:
This is not the nicest solution. IIUC, we should collect only those `freeVariables` that are bound somewhere on the way to the toplevel, minus local functions (which will become toplevel definitions, too).
See also https://github.com/effekt-lang/effekt/pull/209/files#diff-2bc35700065c09eb82c800736c2625a2516ddb669bb20acbfce3d9af8dbca683R71

Reply:
I think this code here might be the reason that `pos/defdef.effekt` is failing for LLVM 🤔